pkg/kp/kv.go

```diff
@@ -294,6 +295,11 @@ func (c consulStore) WatchPods(keyPrefix string, quitChan <-chan struct{}, errCh
 					return
 				case errChan <- util.Errorf("Could not parse pod manifest at %s: %s. Content follows: \n%s", pair.Key, err, pair.Value):
 				}
+				// do not send the partially built list, since we know it's incomplete
+				// and some callers may care about elements being removed
+				// TODO(sjung): those callers should probably be refactored to
+				// use NewKeyWatcher instead
+				continue WATCH_LOOP
```

Comment:
This means an invalid manifest would cause a whole host to stop doing deploys. I'd argue that is a worse situation to be in, especially in cases where we may accidentally already have invalid data in `/intent`. This change seems dangerous to me.